share/completions/zpool.fish

```diff
@@ -37,20 +37,20 @@ function __fish_zpool_using_command # zpool command whose completions are looked
 end
 
 function __fish_zpool_list_used_vdevs -a pool
-    if test -z "$pool"
-        zpool list -H -v | grep -E "^\s" | cut -f2 | grep -x -E -v "(spare|log|cache|mirror|raidz.?)"
-    else
-        zpool list -H -v $pool | grep -E "^\s" | cut -f2 | grep -x -E -v "(spare|log|cache|mirror|raidz.?)"
-    end
+    zpool list -H -v $pool | string replace -rf '^\s' '' | string match -rf '^(spare|log|cache|mirror|raidz.?)'
 end
 
 function __fish_zpool_list_available_vdevs
     if test $OS = 'Linux'
-        find /dev -type b | cut -d'/' -f3
+        find /dev -type b | string replace '/dev/' ''
     else if test $OS = 'FreeBSD'
-        for i in (camcontrol devlist | cut -d "(" -f 2 | cut -d "," -f 1); ls /dev | grep "^$i"; end
+        for i in (camcontrol devlist | cut -d "(" -f 2 | cut -d "," -f 1)
```

Comment:
The FreeBSD code doesn't work without admin privileges:

```
mqudsi@freebsd /u/h/m/fish-shell> camcontrol devlist
camcontrol: couldn't open /dev/xpt0: Permission denied
```

and produces the wrong output with them:

```
mqudsi@freebsd /u/h/m/fish-shell> for i in (sudo camcontrol devlist | cut -d "(" -f 2 | cut -d "," -f 1)
                                      set -l files /dev/*
                                      string replace /dev/ '' -- $files | string match -r "^$i"
                                  end
cd0
pass1
pass2
pass3
```

That entire part can be replaced with a single `sysctla -an kern.disks` which returns all valid vdevs, without the `/dev/` prefix to boot (but separated by spaces):

```
mqudsi@freebsd /u/h/m/fish-shell> sysctl -an kern.disks | string split ' '
cd0
da2
da1
da0
nvd0
```

Reply:
Nice, thanks!